### ISSUE TITLE
[Bug] To many re-renders when DateTime uses "Eurpo/Moscow" timezone

### ISSUE THREAD

**Issue by beta3-z** [gravity-ui/date-components#33]
# How to reproduce
(I put a video and a simple CRA project below)

1. Create a DateTime object with Europe/Moscow timezon
```tsx
const timestamp = new Date('Sun Dec 18 2022 21:14:40 GMT+0300').getTime();
const defaultValue = dateTimeParse(timestamp, { timeZone: 'Europe/Moscow' }))
```
2. Pass the date to a DatePicker `value` prop
```tsx
const [value, setValue] = useState<DateTime | null | undefined>(defaultValue)

return <DatePicker value={value} ... />
```
3. Try to change a year in a TextInput

# Expected
Year is changed and React doesn't throw a error

# Actual result
The "Too many re-renders. React limits the number of renders to prevent an infinite loop" error thrown

# Reproduced in Browsers
Yandex.Browser, Safari, Firefox

# Attachments
## Video
https://github.com/gravity-ui/date-components/assets/7751284/addb5b07-a206-4091-98bd-47a1f0cb4692

## Example project: 
[test-gravity-ui.zip](https://github.com/gravity-ui/date-components/files/13949527/test-gravity-ui.zip)

## Full Code Example
```tsx
import React, {useState} from 'react';
import './App.css';

import { ThemeProvider } from '@gravity-ui/uikit';
import {DatePicker} from '@gravity-ui/date-components';
import {DateTime, dateTimeParse} from '@gravity-ui/date-utils';

function App() {
    const timestamp = new Date('Sun Dec 18 2022 21:14:40 GMT+0300').getTime();
    const defaultValue = dateTimeParse(timestamp, { timeZone: 'Europe/Moscow' })
    const [value, setValue] = useState<DateTime | null | undefined>(defaultValue)

  return (<ThemeProvider theme="light">
        <DatePicker value={value}
                    onUpdate={setValue}
                    format="DD.MM.YYYY HH:mm" />
      </ThemeProvider>
  );
}

export default App;
```

**Comment by ValeraS** [comment]
Thanks for the details on the issue, PR #25 fixes it. We plan to release a major version with new date-utils at the end of January after releasing `@gravity-ui/uikit` v6.

**Comment by ValeraS** [comment]
fixed in #54
